SwiftDraft/ReliantFramework/ReliantFrameworkTests/ReliantFrameworkTests.swift

```diff
@@ -54,17 +54,25 @@ struct ContextNeedingContext : ReliantContext {
         return ContextNeedingContext()
     }
 }
+class SubWaver : Waver {
+    func wave(reason: String) -> String {
+        return "Substitute waving"
+    }
+}
 
-
-
-
+class SubstituteContext : SimpleReferenceContext {
+    override init() {
+        super.init()
+        waver = SubWaver()
```

Comment:
Yeah the current method is pretty much a no-go. I think there are still some alternatives to explore though.